platform/broadcom/sonic-platform-modules-delta/ag5648/scripts/led_control

```diff
@@ -0,0 +1,221 @@
+    if [ "`echo $?`" -eq "0" ]; then
+        psu2_eeprom_num=$((psu2_eeprom_num+1))
+    elif [ "`echo $?`" -eq "2" ]; then
+        psu2_eeprom_num=$((psu2_eeprom_num))
+    fi
+}
+
+catpsufanspeed(){
+
+    psu1_rpm_normal_num=0
+    psu1_rpm=`cat $PSU1_FAN_RPM`
+	
+    if [ "${psu1_rpm}" -ne "960" ] && [ "${psu1_rpm}" -ne "0" ]; then
+        psu1_rpm_normal_num=$((psu1_rpm_normal_num+1))
+    elif [ "${psu1_rpm}" -eq "960" ] || [ "${psu1_rpm}" -eq "0" ]; then
+        psu1_rpm_normal_num=$((psu1_rpm_normal_num))
+    fi
+	
+    psu2_rpm_normal_num=0
+    psu2_rpm=`cat $PSU2_FAN_RPM` 
+	
+    if [ "${psu2_rpm}" -ne "960" ] && [ "${psu2_rpm}" -ne "0" ]; then
+        psu2_rpm_normal_num=$((psu2_rpm_normal_num+1))
+    elif [ "${psu2_rpm}" -eq "960" ] || [ "${psu2_rpm}" -eq "0" ]; then
+        psu2_rpm_normal_num=$((psu2_rpm_normal_num))
+    fi
+
+}
+
+setfanled(){
+    if [ "${fan_eeprom_num}" -eq "4" ] && [ "${fan_rpm_normal_num}" -eq "4" ]; then
+        echo "fan_green" > $LED_CONTROL
+    elif [ "${fan_eeprom_num}" -eq "4" ] && [ "${fan_rpm_normal_num}" -lt "4" ]; then
+        echo "fan_Amber" > $LED_CONTROL
+    elif [ "${fan_eeprom_num}" -gt "0" ]; then
+        echo "pwr_Blinking_yellow" > $LED_CONTROL
+    else
+        echo "fan_off" > $LED_CONTROL
+    fi
+}
+
+setpsuled(){
+    if [ "${psu1_rpm_normal_num}" -eq "1" ] && [ "${psu2_rpm_normal_num}" -eq "1" ]; then
+        echo "pwr_green" > $LED_CONTROL
+    elif [ "${psu1_eeprom_num}" -eq "1" ] && [ "${psu2_eeprom_num}" -eq "1" ]; then
+        echo "pwr_Blinking_Amber" > $LED_CONTROL
+    elif [ "${psu1_eeprom_num}" -eq "0" ] || [ "${psu2_eeprom_num}" -eq "0" ]; then
+        echo "pwr_Amber" > $LED_CONTROL
+    else
+        echo "pwr_off" > $LED_CONTROL
+    fi
+
+}
+
+setfantrayled(){
+    if [ "${fan1_rpm_normal_num}" -eq "1" ]; then
+        echo "fan1_green" > $LED_CONTROL
+    elif [ "${fan1_eeprom_num}" -eq "1" ]; then
+        echo "fan1_Amber" > $LED_CONTROL
+    else
+        echo "fan1_off" > $LED_CONTROL
+    fi
+	
+    if [ "${fan2_rpm_normal_num}" -eq "1" ]; then
+        echo "fan2_green" > $LED_CONTROL
+    elif [ "${fan2_eeprom_num}" -eq "1" ]; then
+        echo "fan2_Amber" > $LED_CONTROL
+    else
+        echo "fan2_off" > $LED_CONTROL
+    fi
+	
+    if [ "${fan3_rpm_normal_num}" -eq "1" ]; then
+        echo "fan3_green" > $LED_CONTROL
+    elif [ "${fan3_eeprom_num}" -eq "1" ]; then
+        echo "fan3_Amber" > $LED_CONTROL
+    else
+        echo "fan3_off" > $LED_CONTROL
+    fi
+	
+    if [ "${fan4_rpm_normal_num}" -eq "1" ]; then
+        echo "fan4_green" > $LED_CONTROL
+    elif [ "${fan4_eeprom_num}" -eq "1" ]; then
+        echo "fan4_Amber" > $LED_CONTROL
+    else
+        echo "fan4_off" > $LED_CONTROL
+    fi
+	
+}
+
+platformstatus(){
+    
+    echo "sys_green" > $LED_CONTROL
+    catfaneeprom
+    catfanspeed
+    setfanled
+    setfantrayled
+	
+    catpsueeprom
+    catpsufanspeed
+    setpsuled	
+}
+
+while true
+do
+    platformstatus
```

Comment:
what if one of the fantray is not inserted?

Reply:
Script will not be break. It changes the color of the led.